frontend/src/components/EllipsisText.vue

```diff
@@ -1,17 +1,35 @@
 <template>
-  <span class="tooltip-wrapper w-full">
-    <span v-if="hasTextOverflow" class="tooltip whitespace-nowrap">
-      <slot></slot>
-    </span>
-    <div ref="textRef" class="truncate w-full"><slot></slot></div>
-  </span>
+  <NEllipsis
+    :line-clamp="lineClamp"
+    :tooltip="{
+      showArrow: false,
+      raw: true,
+      contentStyle: tooltipStyle,
+    }"
+  >
+    <slot></slot>
+  </NEllipsis>
 </template>
 
 <script lang="ts" setup>
-import { ref, computed } from "vue";
+import { NEllipsis } from "naive-ui";
 
-const textRef = ref<HTMLElement | null>(null);
-const hasTextOverflow = computed(() => {
-  return textRef.value && textRef.value.offsetWidth < textRef.value.scrollWidth;
+defineProps({
+  lineClamp: {
+    type: Number,
+    default: 1,
+  },
 });
+
+const tooltipStyle = `
+  border-radius: 0.25rem;
+  background-color: rgb(0 0 0 / var(--tw-bg-opacity));
+  --tw-bg-opacity: 0.75;
+  padding-left: 0.5rem;
+  padding-right: 0.5rem;
+  padding-top: 0.25rem;
+  padding-bottom: 0.25rem;
+  --tw-text-opacity: 1;
+  color: rgb(255 255 255 / var(--tw-text-opacity));
+`;
```

Comment:
Is that possible to use tailwindcss classes rather than css styles here?

Reply:
No, [it](https://www.naiveui.com/en-US/light/components/popover) only accepts style strings rather than class names.

![CleanShot 2022-09-01 at 11 36 28@2x](https://user-images.githubusercontent.com/56376387/187826455-e32914eb-be2e-4833-b018-6e71bb124761.png)